[Mooncake] Config: Use extra config in LMCacheEngine

stmatengss opened the pull request:
FILL IN THE PR DESCRIPTION HERE

USE extra_config in LMCacheEngineConfig to generate MooncakeConfig. It can reduce extra configuration files.
It is a follow-up of #747 
PTAL @maobaolong @YaoJiayi 

**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>
<!-- inside this <details> section, markdown rendering does not work, so we use raw html here. -->
<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to LMCache! Before submitting the pull request, please ensure the PR meets the following criteria. This helps us maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Please try to classify PRs for easy understanding of the type of changes. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core LMCache logic (e.g., <code>LMCacheEngine</code>, <code>Backend</code> etc.)</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li> Please include sufficient tests to ensure the change is stay correct and robust. This includes both unit tests and integration tests.</li>
</ul>

<h3>What to Expect for the Reviews</h3>

To create a new tag for lmcache (Note: `v` prefix is required):
`git tag vx.x.x`
`git push origin vx.x.x` (same version again)

For example:
`git tag v0.3.0`
`git push origin v0.3.0`

In case the workflow fails, delete the tag and try again:
`git tag -d vx.x.x`
`git push origin :refs/tags/vx.x.x`

For example:
`git tag -d v0.3.0`
`git push origin :refs/tags/v0.3.0`

To create a new release and publish `lmcache` Python package to PyPi:
`git remote add upstream git@github.com:LMCache/LMCache.git`
`gh release create vx.x.x --repo LMCache/LMCache --title "vx.x.x" --notes "<Add description>"`

For example:
`git remote add upstream git@github.com:LMCache/LMCache.git`
`gh release create v0.3.0 --repo LMCache/LMCache --title "v0.3.0" --notes "LMCache v0.3.0 is a feature release. Users are encouraged to upgrade for the best experience."`

> [!TIP]
> The creation of a release and subsequent tag generation can be done alternatively from the LMCache [releases](https://github.com/LMCache/LMCache/releases) page.

We aim to address all PRs in a timely manner. If no one reviews your PR within 5 days, please @-mention one of KuntaiDu, ApostaC or YaoJiayi.

vllmellm committed:
[Enhancement] Support for ROCm on LMCache (#702)

* enable compiling with rocm

* initial hipified files

* update hip code

* add documentation

* add no op _ldg as in vLLM

Signed-off-by: tjtanaa <tunjian.tan@embeddedllm.com>

* fix block ids for image requests

* fix hipify bugs

* add hipify step to setup; better naming

* update hipify docs

* prefer USE_ROCM; prefer LMCACHE_LDG

* clean up

* clean up

* minor bug fix; remove hipify file

* remove hipified files

* remove hipified files

* remove hipify notes; fix minor bug

* revert

* fix bug

* clean up setup.py; add hipified files to git ignore

* clean up

* fix format; remove unnecessary condition

Signed-off-by: vllmellm <vllm.ellm@embeddedllm.com>

---------

Signed-off-by: tjtanaa <tunjian.tan@embeddedllm.com>
Signed-off-by: vllmellm <vllm.ellm@embeddedllm.com>
Co-authored-by: noobhappylife <aratar1991@hotmail.com>
Co-authored-by: tjtanaa <tunjian.tan@embeddedllm.com>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

chunxiaozheng committed:
[bugfix] set gds_path by defaults (#798)

Signed-off-by: idellzheng <idellzheng@tencent.com>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

stmatengss committed:
Config: Use extra config in LMCacheEngine

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

stmatengss committed:
Use pre-commit for formatting

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

maobaolong commented on lmcache/v1/storage_backend/connector/mooncakestore_connector.py:
```diff
@@ -74,6 +75,25 @@ def load_from_env() -> "MooncakeStoreConfig":
+        if extra_config is None:
+            raise ValueError("The extra config is not set.")
+        return MooncakeStoreConfig(
+            local_hostname=str(extra_config["local_hostname"]),
```
Are these arguments all necessary? If some of them is optional, maybe we can supply the default value if absent?

stmatengss replied:
Sure.

stmatengss committed:
add default values

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

YaoJiayi reviewed (commented):
LGTM! @stmatengss Could you fix the DCO?

YaoJiayi committed:
[Enhacement] Improve layerwise cache store/load (#794)

* improve log

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* optimize layerwise

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* further optimize

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* fix format

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* fix test

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* add more tests

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* fix unit test

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

* add more tests

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>

---------

Signed-off-by: YaoJiayi <120040070@link.cuhk.edu.cn>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

stmatengss commented:
> LGTM! @stmatengss Could you fix the DCO?

Done. THX!

chunxiaozheng committed:
[optimize] support don't save unfull chunk (#804)

Signed-off-by: idellzheng <idellzheng@tencent.com>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

chunxiaozheng committed:
[bugfix] set use_mla in lmcache metadata (#810)

Signed-off-by: idellzheng <idellzheng@tencent.com>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

maobaolong committed:
[#771]Support dynamic loading of external remote connector implementations (#774)

* Support dynamic loading of external remote connector implementations

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

* Add example for external remote connector

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

* Remove useless config

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

---------

Signed-off-by: baoloongmao <baoloongmao@tencent.com>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

xiaguan committed:
feat: enhance controller manager to support JSON messages from Mooncake (#799)

- Add comprehensive documentation for message handling
- Support both JSON and MessagePack formats for better interoperability
- Improve code readability with cleaner message routing logic
- Enable integration with external systems like Mooncake for event notifications

Example JSON message format:
{
  "type": "KVAdmitMsg",
  "instance_id": "lmcache_default_instance",
  "worker_id": 0,
  "key": "vllm@/root/.cache/modelscope/hub/models/...",
  "location": "mooncake_cpu_ram"
}

Signed-off-by: Jinyang Su <751080330@qq.com>
Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

maobaolong commented on lmcache/v1/storage_backend/local_cpu_backend.py:
```diff
@@ -65,6 +65,7 @@ def __init__(
         self.lookup_server = lookup_server
         self.memory_allocator = memory_allocator
         self.lmcache_worker = lmcache_worker
+        self.config = config
```
Is this necessary?

stmatengss replied:
Yes. This is the only way the Mooncake Connector can get its configuration from the upper layer.

maobaolong replied:
Maybe you can pass the `LMCacheEngineConfig` within `CreateConnector` method by add a new argument for init method of the `MooncakestoreConnector`. Therefore, we can avoid change the unrelated class `LocalCPUBackend`. Any thoughts?

stmatengss replied:
Done

stmatengss committed:
fetch config in connector factory

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

stmatengss committed:
pre-commit

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

stmatengss committed:
Merge branch 'dev' into mooncake_extra_config

hickeyma committed:
[CI]: Add static checker for GitHub actions workflows (#808)

* [CI]: Add linting for GitHub actions workflows

Helps improve the quality of actions in the workflows.

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

* Lint fixes

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

---------

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

Shaoting-Feng commented on lmcache/v1/storage_backend/connector/__init__.py:
```diff
@@ -182,6 +182,8 @@ def CreateConnector(
                 host, port, device_name, loop, local_cpu_backend
             )
+            if config is not None:
+                connector.get_config_from_lmcache(config)
```
You first (1) initiated a MooncakestoreConnector and then (2) called `get_config_from_lmcache` function. In Step (1), when you called `self.config = MooncakeStoreConfig.load_from_lmcache_config`, the `self.lmcache_config` hasn't been passed yet. Please correct me if I am wrong.

stmatengss replied:
That's correct; I don't want to add an extra parameter, so let me change it another way.

stmatengss committed:
refactor

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

stmatengss committed:
Merge branch 'dev' into mooncake_extra_config

maobaolong commented on lmcache/v1/storage_backend/connector/mooncakestore_connector.py:
```diff
@@ -96,7 +117,16 @@ def __init__(
+                self.config = MooncakeStoreConfig.from_file(config_file_path)
+            elif lmcache_config is not None:
+                self.config = MooncakeStoreConfig.load_from_lmcache_config(
+                    lmcache_config.extra_config
```
The passed parameter type is not match the defined parameter type.

stmatengss replied:
Sorry for omitting. Fix it.

maobaolong reviewed (commented):
Left a single comment.

stmatengss committed:
fix

Signed-off-by: Teng Ma <sima.mt@alibaba-inc.com>

maobaolong reviewed (approved):
LGTM, thanks for your contribution!

stmatengss commented:
An example in YAML is here:
```
extra_config:
  local_hostname: "localhost"
  metadata_server: "http://ip:8080/metadata"
  protocol: "rdma"
  device_name: ""
  global_segment_size: 10737418240
  master_server_address: "ip:50051"
  local_buffer_size: 5368709120
```

maobaolong commented:
@YaoJiayi would you like to take a look at this PR？ Thanks!

Shaoting-Feng reviewed (approved):
LGTM

stmatengss committed:
Merge branch 'dev' into mooncake_extra_config
